hello-linux/hello.c

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright(C) 2017 Pedro H. Penna <pedrohenriquepenna@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+
+/**
+ * @brief NoC major device number.
+ */
+#define NOC_MAJOR 253
+
+/**
+ * @brief NoC minor device number.
+ */
+#define NOC_MINOR 0
+
+/**
+ * @brief NoC device filename.
+ */
+const char *devname = "/noc";
```

Comment:
@imphil actually there is no need for that, because we can create special inodes whereever we want. Initially I created under the root directory to keep INITRAMFS as simple as possible. However, now that Linux supports /dev I'll move the special inode to there.